model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -335,6 +335,12 @@ class DependencyGraphBuilder<D>(
         val dependencies2 = dependencies.associateBy { dependencyHandler.identifierFor(it) }
         if (!dependencies2.keys.containsAll(dependencies1)) return false
 
+        if (dependencyHandler::class.simpleName == "GradleDependencyHandler") {
+            // In case of Gradle, we can skip the costly deep comparison.
```

Comment:
Please use passive voice everywhere, i.e. no "we", "I", "our" etc. but instead say

> In case of Gradle, the costly deep comparison ca be skipped.